### PR TITLE
add in synclogs

### PR DESCRIPTION
**Pull request by snopoke**
@czue @NoahCarnahan @benrudolph 

Not done yet but open for review

### CODE REVIEW

File: prototype/mobile_endpoint/dao.py [hunk 1]
```diff
@@ -72,27 +72,21 @@ def get_reverse_indexed_cases(self, domain, case_ids):
...
 class SQLDao(AbsctractDao):
-    def commit(self, xform, cases):
-        def to_sql(doc):
-            if isinstance(doc, XFormInstance):
-                return cls_for_doc_type(doc.doc_type).from_generic(doc)
-            elif isinstance(doc, CommCareCase):
-                return CaseData.from_generic(doc)
+    def commit(self, xform, case_result):
+        cases = case_result.cases
+        synlog = case_result.synclog
```
Comment (czue) [inline prototype/mobile_endpoint/dao.py]: tiny, but there's a typo here

**Comment by czue** [comment]
looked quickly and looks good. we might want to explore other possible representations that don't require saving the entire list of cases every time, but that can be changed later

**Comment by snopoke** [comment]
Fully agree. From what I've read you can append to array's in postgresql
but you can't remove elements from them without re-writing the hole array.
Initially this info was in a join table but that didn't seem very
efficient. We'd also have to change the receiver code to keep track of
elements added / removed from the lists.

**Comment by snopoke** [comment]
Added some basic tests. I think this is GTG

**Pull request merged**